core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -2944,6 +3022,18 @@ private void registerQuery(
           enclosingNode,
           null,
           false);
+
+      SqlNode condition = ((SqlInsert) node).getCondition();
+      if (condition != null) {
+        SqlValidatorScope insertScope = getWhereScope((SqlSelect) insertCall.getSource());
+        requireNonNull(condition, "insertCall.getSource()");
```

Comment:
I'm confused by this check if the condition in the if-clause is already `condition != null`

Reply:
Good catch!